course/functions.php

```diff
@@ -96,4 +98,56 @@ function course_register_block_patterns_category() {
 	);
 }
 
-add_action( 'init', 'course_register_block_patterns_category' );
\ No newline at end of file
+add_action( 'init', 'course_register_block_patterns_category' );
+
+/**
+ * Determine the theme variation and save in option.
+ */
+function course_save_global_styles( $post_id, $post, $update ) {
+	if ( 'wp_global_styles' !== $post->post_type ) {
+		return;
+	}
+
+	$global_styles      = json_decode( $post->post_content, true );
+	$global['settings'] = isset( $global_styles['settings'] ) ? $global_styles['settings'] : array();
+	$global['styles']   = isset( $global_styles['styles'] ) ? $global_styles['styles'] : array();
+	$variations         = WP_Theme_JSON_Resolver::get_style_variations();
+	$current_variation  = 'default';
+	foreach ( $variations as $variation ) {
+		if ( $variation['settings'] === $global['settings'] && $variation['styles'] === $global['styles'] ) {
+			$current_variation = strtolower( $variation['title'] );
+		}
+	}
+	update_option( 'course_theme_variation', $current_variation );
+}
+
+add_action( 'save_post', 'course_save_global_styles', 10, 3 );
+
+/**
+ * Delete the theme variation option when the global styles post is deleted.
+ */
+function course_delete_global_styles( $post_id, $post ) {
+	if ( 'wp_global_styles' !== $post->post_type ) {
+		return;
+	}
+
+	delete_option( 'course_theme_variation' );
+}
+
+add_action( 'delete_post', 'course_delete_global_styles', 10, 2 );
+
+/**
+ * Add the theme variation to the body class.
+ */
+function course_add_variation_body_class( $classes ) {
+	$current_variation = get_option( 'course_theme_variation' );
+	if ( ! $current_variation ) {
+		return $classes;
+	}
+
+	$classes[] = 'course-variation-' . $current_variation;
+
+	return $classes;
+}
+
+add_action( 'body_class', 'course_add_variation_body_class' );
```

Comment:
I think we'll also need to add this class to the editor so that we can apply styles there too; however, I think this might be a bit tricky so it may be better to create a separate card for it so that we don't hold this one up.

Reply:
Created a separate card for that: https://github.com/Automattic/sensei/issues/6852